shared/util/make-reducer.tsx

```diff
@@ -0,0 +1,22 @@
+import {TypedActionsMap, TypedActions} from '../actions/typed-actions-gen'
+import {produce, Draft} from 'immer'
+
+type GetTypes<A> = A extends {type: string} ? A['type'] : never
+
+function makeReducer<A, S>(
+  initialState: S,
+  map: {
+    [type in GetTypes<A>]?: type extends keyof TypedActionsMap
+      ? ((state: Draft<S>, action: TypedActionsMap[type]) => void | S)
+      : never
+  }
+) {
+  return (state: S = initialState, action: TypedActions): S =>
+    // @ts-ignore
```

Comment:
Does this ignore affect typing of the reducers that use it?  Why's it here?

Reply:
We say what the type is explicitly so it doesn't matter. See the original implementation for context